### PR TITLE
Generate type declaration at build time to support TypeScript

### PR DESCRIPTION
**Pull request by roackb2**
This PR add tsconfig file and generate type declaration files to support TypeScript.

### CODE REVIEW

**Comment by marina-mosti** [comment]
Hey @roackb2 thanks for the PR :)

@logaretm mind taking a peek?

File: dist/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export default function VeeValidatePlugin(opts: any): (baseReturns: any) => any;
+export function withField(el: any): any;
+export function mapElementsInSchema(schema: any[], fn: Function): any[];
```
Comment (logaretm) [inline dist/index.d.ts:3]: These are probably exported by mistake, I didn't realize that till now.

Could we remove those `export` statements from the main `index.js` file? They shouldn't be used and there shouldn't be any named export when we export the plugin function by default.

**Review by logaretm — CHANGES_REQUESTED** [review]
Thanks, this looks awesome but I have a minor change to request here.

File: dist/index.d.ts (hunk 1 above)
Comment (roackb2) [reply dist/index.d.ts]: Hi @logaretm,

I removed export of these two functions and build again.
Would you please take a look and see if this is the desired result?

**Review by logaretm — APPROVED** [review]

**Comment by marina-mosti** [comment]
Looks good @logaretm, but we need to port to monorepo

@roackb2 we have two ways we can do this :)
You could please A) open a new PR in the main formvuelate repository with the new monorepo structure so that when we merge you conserve credit of contribution on the main repo
or B) we merge this and then take care of copying it over

WDYT?

**Comment by logaretm** [comment]
@roackb2 Your work has been moved to [here](https://github.com/formvuelate/formvuelate/pull/215) to better function with the monorepo structure.

You've been credited as a co-author, I will be closing this since this repo will be archived in favor of the mono-repo [here](https://github.com/formvuelate/formvuelate)

**Comment by roackb2** [comment]
@logaretm Thanks for your help!

**Pull request closed without merging**